rust/ql/lib/ext/generated/rust/lang-alloc.model.yml

```diff
@@ -18,14 +18,15 @@ extensions:
       - ["lang:alloc", "<crate::borrow::Cow as crate::ops::arith::AddAssign>::add_assign", "Argument[0]", "Argument[self].Reference", "value", "dfc-generated"]
       - ["lang:alloc", "<crate::borrow::Cow as crate::ops::deref::Deref>::deref", "Argument[self].Reference.Field[crate::borrow::Cow::Borrowed(0)]", "ReturnValue", "value", "dfc-generated"]
       - ["lang:alloc", "<crate::borrow::Cow>::into_owned", "Argument[self].Field[crate::borrow::Cow::Owned(0)]", "ReturnValue", "value", "dfc-generated"]
-      - ["lang:alloc", "<crate::borrow::Cow>::to_mut", "Argument[self].Reference.Field[crate::borrow::Cow::Owned(0)]", "ReturnValue", "value", "dfc-generated"]
+      - ["lang:alloc", "<crate::borrow::Cow>::to_mut", "Argument[self].Reference.Field[crate::borrow::Cow::Owned(0)]", "ReturnValue.Reference", "value", "dfc-generated"]
```

Comment:
When should the qualifier be `Argument[self].Reference` vs just `Argument[self]`?  I thought `self` is always a reference?

Reply:
When a method takes `self` without `&` or `&mut` (for instance [Option::unwrap](https://doc.rust-lang.org/std/option/enum.Option.html#method.unwrap)) then `self` is not a reference but passed by value.

So roughly, `Argument[self].Reference` corresponds to `&self` and `&mut self`, and `Argument[self]` corresponds to `Argument[self]`.

Here `Cow::to_mut` takes `&mut self` and [dereferences the `self` pointer](https://doc.rust-lang.org/src/alloc/borrow.rs.html#273) which gives rise to `Argument[self].Reference`.